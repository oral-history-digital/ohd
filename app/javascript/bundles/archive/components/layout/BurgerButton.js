import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function BurgerButton({ open, onClick }) {
    return (
        <button type="button"
                className={classNames('BurgerButton', {
                    'is-open': open,
                    'is-closed': !open,
                })}
                onClick={onClick}>
            <i className={classNames('BurgerButton-icon', 'fa', {
                'fa-close': open,
                'fa-bars': !open,
            })}/>
        </button>
    );
}

BurgerButton.propTypes = {
    open: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
};

BurgerButton.defaultProps = {
    open: false,
};

export default BurgerButton;
