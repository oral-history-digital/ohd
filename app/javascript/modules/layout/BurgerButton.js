import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FaTimes, FaBars } from 'react-icons/fa';

function BurgerButton({
    className,
    open,
    onClick,
}) {
    return (
        <button
            type="button"
            className={classNames(className, 'Button', 'BurgerButton', {
                'is-open': open,
                'is-closed': !open,
            })}
            onClick={onClick}
        >
            {
                open ?
                    <FaTimes className="BurgerButton-icon" /> :
                    <FaBars className="BurgerButton-icon" />
            }
        </button>
    );
}

BurgerButton.propTypes = {
    className: PropTypes.string,
    open: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
};

BurgerButton.defaultProps = {
    open: false,
};

export default BurgerButton;
