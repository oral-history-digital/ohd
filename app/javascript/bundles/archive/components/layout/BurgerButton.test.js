import React from 'react';
import renderer from 'react-test-renderer';
import BurgerButton from './BurgerButton';

it('renders correctly in closed state', () => {
    const tree = renderer
        .create(<BurgerButton onClick={() => {}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly in open state', () => {
    const tree = renderer
        .create(<BurgerButton open onClick={() => {}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
