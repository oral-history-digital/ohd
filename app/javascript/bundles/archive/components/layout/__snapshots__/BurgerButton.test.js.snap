// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly in closed state 1`] = `
<button
  className="BurgerButton is-closed"
  onClick={[Function]}
  type="button"
>
  <i
    className="BurgerButton-icon fa fa-bars"
  />
</button>
`;

exports[`renders correctly in open state 1`] = `
<button
  className="BurgerButton is-open"
  onClick={[Function]}
  type="button"
>
  <i
    className="BurgerButton-icon fa fa-close"
  />
</button>
`;
